File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/store/meta/RendezvousStoreSetResponse.scala
package com.webank.eggroll.core.deepspeed.store.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions


case class RendezvousStoreSetResponse()

object RendezvousStoreSetResponse {
  implicit def serialize(src: RendezvousStoreSetResponse): Array[Byte] = {
    val builder = Deepspeed.StoreSetResponse.newBuilder()
    builder.build().toByteArray
  }

  implicit def deserialize(byteString: ByteString): RendezvousStoreSetResponse = {
    val proto = Deepspeed.StoreSetResponse.parseFrom(byteString)
    RendezvousStoreSetResponse()
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/deepspeed/store/RendezvousStoreSetResponse.java
package com.eggroll.core.deepspeed.store;

import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.Deepspeed;

public class RendezvousStoreSetResponse {
    public static byte[] serialize(RendezvousStoreSetResponse src) {
        Deepspeed.StoreSetResponse.Builder builder = Deepspeed.StoreSetResponse.newBuilder();
        return builder.build().toByteArray();
    }

    public static RendezvousStoreSetResponse deserialize(byte[] bytes) throws InvalidProtocolBufferException {
        Deepspeed.StoreSetResponse proto = Deepspeed.StoreSetResponse.parseFrom(bytes);
        return new RendezvousStoreSetResponse();
    }
}